crates/oxc_macros/src/declare_all_lint_rules.rs

```diff
@@ -0,0 +1,52 @@
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::parse::{Parse, ParseStream};
+use syn::Result;
+
+pub struct AllLintRulesMeta {
+    paths: Vec<syn::Path>,
+}
+
+impl Parse for AllLintRulesMeta {
+    fn parse(input: ParseStream<'_>) -> Result<Self> {
+        let paths = input
+            .parse_terminated::<syn::Path, syn::Token![,]>(syn::Path::parse)?
+            .into_iter()
+            .collect();
+
+        Ok(Self { paths })
+    }
+}
+
+fn define_rule_mod(path: &syn::Path) -> TokenStream {
+    let mut segments = path.segments.iter().rev().peekable();
+    let first = &segments.next().unwrap().ident;
+    let mut stmts = quote! {mod #first;};
+    if segments.peek().is_some() {

```

Comment:
very nit:
is there a need for the duplication?

Reply:
we can reduce more code if it's doable

Reply:
> very nit:
> is there a need for the duplication?

Not nitpicking at all, this bothered me, I looked for a cleaner way to achieve:
- `pub mod` for nested `mod` statements
- `;` at the end of the inner most `mod` statement
- no `pub` for the top-level `mod` statement

Somehow this is the best I could come up with, very open to suggestions!